luigi/contrib/pyspark_runner.py

```diff
@@ -36,11 +36,66 @@
 import sys
 import os
 
+from luigi import configuration
+
 # this prevents the modules in the directory of this script from shadowing global packages
 sys.path.append(sys.path.pop(0))
 
 
-class PySparkRunner(object):
+class _SparkEntryPoint(object):
+    def __init__(self, conf):
+        self.conf = conf
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+
+class SparkContextEntryPoint(_SparkEntryPoint):
+    sc = None
+
+    def __enter__(self):
+        from pyspark import SparkContext
+        self.sc = SparkContext(conf=self.conf)
+        return self.sc, self.sc
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.sc.stop()
+
+
+class SparkSessionEntryPoint(_SparkEntryPoint):
+    spark = None
+
+    def _check_major_spark_version(self):
+        from pyspark import __version__ as spark_version
+        major_version = int(spark_version.split('.')[0])
+        if major_version < 2:
+            raise RuntimeError(
+                '''
+                Apache Spark {} does not support SparkSession entrypoint.
+                Try to set 'pyspark_runner.use_spark_session' to 'False' and switch to old-style syntax
+                '''.format(spark_version)
+            )
+
+    def __enter__(self):
+        self._check_major_spark_version()
+        from pyspark.sql import SparkSession
+        self.spark = SparkSession \
+            .builder \
+            .config(conf=self.conf) \
+            .enableHiveSupport() \
```

Comment:
Does it make sense to always enable Hive Support? Was that a default for Spark 1.x SparkContexts?

Reply:
@dlstadther 
In spark 1.x hive support was provided in `HiveContext` - object differing from `SparkContext`
This object was obtained and managed inside client's `PySparkTask` inheritor
Now this two objects are merged into single `SparkSession`
But the fact is that when we create a `SparkSession` object once (inside of `SparkSessionEntryPoint`) clients are not able to enableHiveSupport in their client code since session is already started.
I think we can add a bool field `enable_hive_support` into `SparkSessionEntryPoint` which could be passed from `PySparkTask` inheritor

Reply:
Got it. I'm ok starting with it enabled and someone can disable in the future if desired.